src/Features/Core/Portable/CodeStyle2/ExampleCodeStyleProvider.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeStyle2
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    class ExampleDiagnosticAnalyzer : ExampleCodeStyleProvider.DiagnosticAnalyzer
+    {
+    }
+
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = "Example code refactoring provider")]
+    class ExampleCodeRefactoringProvider : ExampleCodeStyleProvider.CodeRefactoringProvider
+    {
+    }
+
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = "Example code fix provider")]
+    class ExampleCodeFixProvider : ExampleCodeStyleProvider.CodeFixProvider
+    {
+    }
```

Comment:
These are the three classes a feature would still have to export. Note they are very intentionally empty. They do not need to do anything, and literally exist solely so we can add the export attributes somewhere so these can be found by the relevant subsystems.

in an ideal world we could just put these attributes on the CodeStyleProvider. however, since each of these attributes expects a particular inheritance hierarchy, and because they all use classes, there is no way to make CodeStyleProvider extend all three of these classes.

Alternatively, we could just have something like ExportCodeStyleProvider, and have the relevant subsystems know how to import that. Then, the CodeStyleProvider would expose the 3 separate providers from itself. That would be trivial to do for CodeRefactoringProvider/CodeFixProvider. however, it's unclear to me how we would fit that into the DiagnosticAnalyzer infrastructure.